ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-NursingIntervention.xsl

```diff
@@ -0,0 +1,169 @@
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
+-->
+
+<xsl:stylesheet exclude-result-prefixes="#all"
+    xmlns="http://hl7.org/fhir"
+    xmlns:util="urn:hl7:utilities" 
+    xmlns:f="http://hl7.org/fhir" 
+    xmlns:xd="http://www.oxygenxml.com/ns/doc/xsl"
+    xmlns:nf="http://www.nictiz.nl/functions" 
+    xmlns:nm="http://www.nictiz.nl/mappings"
+    xmlns:uuid="http://www.uuid.org"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema" 
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform" 
+    version="2.0">
+    
+    <xsl:output method="xml" indent="yes"/>
+    <xsl:strip-space elements="*"/>
+    
+    <xd:doc scope="stylesheet">
+        <xd:desc>Converts ada verpleegkundige_interventie to FHIR CarePlan conforming to profile nl-core-NursingIntervention</xd:desc>
+    </xd:doc>
+    
+    <xd:doc>
+        <xd:desc>Create an nl-core-NursingIntervention as a CarePlan FHIR instance from ada verpleegkundige_interventie element.</xd:desc>
+        <xd:param name="in">ADA element as input. Defaults to self.</xd:param>
+        <xd:param name="subject">Optional ADA instance or ADA reference element for the patient.</xd:param>
+    </xd:doc>
+    <xsl:template match="verpleegkundige_interventie" name="nl-core-NursingIntervention" mode="nl-core-NursingIntervention" as="element(f:CarePlan)">
+        <xsl:param name="in" select="." as="element()?"/>
+        <xsl:param name="subject" select="patient/*" as="element()?"/>
+        <xsl:param name="requester" select="aanvrager/*" as="element()?"/>
+        <xsl:param name="addresses" select="indicatie/*" as="element()?"/>
+        <xsl:param name="goal" select="behandeldoel/*" as="element()?"/>
+        <xsl:param name="medicalDevice" select="medisch_hulpmiddel/*" as="element()?"/>
+        <xsl:param name="performer" select="(uitvoerder/zorgverlener/*, uitvoerder/verzorger/*, uitvoerder/patient/*)[1]" as="element()?"/>
+        
+        <xsl:for-each select="$in">
+            <CarePlan>
+                <xsl:call-template name="insertLogicalId"/>
+                <meta>
+                    <profile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-NursingIntervention"/>
+                </meta>
+                <status>
+                    <xsl:choose>
+                        <xsl:when test="xs:date(actie_eind_datum_tijd/@value) lt current-date()">
+                            <xsl:attribute name="value" select="'completed'"/>
+                        </xsl:when>
+                        <xsl:otherwise>
+                            <!-- In any other situation, we cannot know if the status is 'preparation' or 'in-progress' because there is no concept of the time difference between transformation and sending the FHIR resource -->
```

Comment:
_preparation_ and _in-progress_ aren't CarePlan statusses. I think it makes sense to call it _active_ by default -- when communicating this, it's not a _draft_ and the other statuses don't apply.

Reply:
Woops, copied the comment from elsewhere.

But besides that, I would refrain from using any other status than completed or unknown. We just cannot know for sure if the resource is still active at the moment of transfer. This is in line with STU3 mappings (and other mappings somewhere in this branch, but these can be changed ofcourse).

Reply:
Well, we should think about this for general guidance in the future. For NursingIntervention the only use case I know is to communicate the current treatment policy, which translates to an active plan. And for the mappings the two use cases that I know of are to generate examples and to generate qualification materials. So from that perspective, _active_ seems like a safe bet.

If there are other use cases for the mappings that require another status ... well that's not something I think we need to take into account at the moment.